libraries/lib-stretching-sequence/AudioSegment.h

```diff
@@ -0,0 +1,22 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  AudioSegment.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+
+#pragma once
+
+#include "SampleCount.h"
+
+class AudioSegment
+{
+public:
+   virtual ~AudioSegment() = default;
```

Comment:
I would define `AudioSegment.cpp` just to define the destructor non-inline -- so that the linker can avoid de-duplication of the definition of the destructor and the virtual function table of the class; also so that "AudioSegment.h" is included in at least one translation unit before any other header, and we can be sure the header does not require any earlier #include for it to compile.

Reply:
I don't know of evidence that the compilation time gain is worth the manual effort. The early-include verification... yeah, all right.

Reply:
Oops : I added `AudioSegment.cpp` and moved the dtor definition there, and now I'm getting unresolved external symbol. I can't see why, so at least for now I'll leave it with the inline definition.